tests/Backend/Workspaces/MetadataFromSnowflakeWorkspaceTest.php

```diff
@@ -19,6 +19,129 @@ public function setUp()
         }
     }
 
+    public function testIncrementalLoadOnlyUpdateDataTypeLengthOnlyUpward()
+    {
+        // create workspace and source table in workspace
+        $workspaces = new Workspaces($this->_client);
+        $workspace = $workspaces->createWorkspace(["backend" => "snowflake"]);
+        $connection = $workspace['connection'];
+        $db = $this->getDbConnection($connection);
+        $db->query("create table \"test.metadata_columns\" (
+                    \"id\" varchar(16),
+                    \"name\" varchar(16)
+                );");
+
+        $tableId = $this->_client->createTableAsyncDirect($this->getTestBucketId(self::STAGE_IN), [
+            'name' => 'metadata_columns',
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => 'test.metadata_columns',
+        ]);
+
+        $expectedNameMetadata = [
+            'KBC.datatype.type' => 'TEXT',
+            'KBC.datatype.nullable' => '1',
+            'KBC.datatype.basetype' => 'STRING',
+            'KBC.datatype.length' => '16',
+            'KBC.datatype.default' => '',
+        ];
+
+        $expectedIdMetadata = [
+            'KBC.datatype.type' => 'TEXT',
+            'KBC.datatype.nullable' => '1',
+            'KBC.datatype.basetype' => 'STRING',
+            'KBC.datatype.length' => '16',
+            'KBC.datatype.default' => '',
+        ];
+
+        $table = $this->_client->getTable($tableId);
+
+        $this->assertEquals([], $table['metadata']);
+
+        $this->assertArrayHasKey('id', $table['columnMetadata']);
+        $this->assertMetadata($expectedIdMetadata, $table['columnMetadata']['id']);
+        $this->assertArrayHasKey('name', $table['columnMetadata']);
+        $this->assertMetadata($expectedNameMetadata, $table['columnMetadata']['name']);
+
+        $db->query("create or replace table \"test.metadata_columns\" (
+                    \"id\" varchar(16),
+                    \"name\" varchar(1)
+                );");
+
+        $this->_client->writeTableAsyncDirect($tableId, [
+            'incremental' => true,
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => 'test.metadata_columns',
+        ]);
+
+        $table = $this->_client->getTable($tableId);
+
+        $this->assertEquals([], $table['metadata']);
+
+        $this->assertArrayHasKey('id', $table['columnMetadata']);
+        $this->assertMetadata($expectedIdMetadata, $table['columnMetadata']['id']);
+        $this->assertArrayHasKey('name', $table['columnMetadata']);
+        $this->assertMetadata($expectedNameMetadata, $table['columnMetadata']['name']);
+
+        //only incremental load doesn't update datatype length down
```

Comment:
tady ten komentář bych otočil. `full load will update datatype length` a k tomu inkrementálnímu bych napsal něco jako `incremental load will not update datatype length as length in workspace is lower than in table and data in table would be truncated`